rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,846 @@
+        <rule id="websphere-mq-02501">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQe API">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"/>
+                </classification>
+            </perform>
+        </rule>
+
+        <!-- Example code to match:
+            Properties env = new Properties();
+            env.put(
+                Context.INITIAL_CONTEXT_FACTORY,
+                "com.ibm.mq.jms.context.WMQInitialContextFactory");
+        -->
+        <rule id="websphere-mq-03000">
+            <when>
+                <!-- WINDUP-615
+                <or>
+                    <javaclass references="com.ibm.mq.jms.context.WMQInitialContextFactory" >
+                        <location>STRING</location>
+                    </javaclass>
+                    <filecontent filename="{*}.properties" pattern="{*}com.ibm.mq.jms.context.WMQInitialContextFactory{*}"/>
+                    <filecontent filename="{*}.xml" pattern="{*}&quot;com.ibm.mq.jms.context.WMQInitialContextFactory&quot;{*}"/>
+                </or>
+                -->
+                <filecontent filename="{*}.{ext}" pattern="{*}com.ibm.mq.jms.context.WMQInitialContextFactory{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS InitialContextFactory">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
+                    <message>For EAP 6, migrate to JBoss HornetQ.</message>
+                    <link description="Migration Guide - HornetQ" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html-single/Migration_Guide/index.html#Migrate_Your_Application_to_Use_HornetQ_as_the_JMS_Provider"/>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+            <where param="ext"><matches pattern="(java|properties|xml)"/></where>
+        </rule>
+
+        <!-- .scp config files - WINDUPRULE-1 -->
+        <rule id="websphere-mq-04000">
+            <when>
+                <filecontent filename="{*}.scp" pattern="{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS configuration file">
+                </classification>
+                <hint effort="3">
+                    <message>The recommended method of configuring messaging in JBoss EAP 6 is in either the Management Console or Management CLI.</message>
```

Comment:
Good, but this hint needs to be a bit more verbose:

"WebSphere `.scp` files are used to ABC and XYZ. these need to be migrated to JBoss EAP 6 JMS using the Management Console or Management CLI."